File: 240123/작은 구슬의 이동/small-marble-movement.py
n, t = tuple(map(int, input().split()))
r, c, d = input().split()
r, c = int(r), int(c)

drs = [-1, 0, 1, 0]
dcs = [0, 1, 0, -1]

mapper = {
    'U': 0,
    'R': 1,
    'D': 2,
    'L': 3,
}

dir_num = mapper[d]


def in_range(r, c):
    return r >= 1 and r <= n and c >= 1 and c <= n

def func():
    global r, c, dir_num

    for _ in range(t):
        nr, nc = r + drs[dir_num], c + dcs[dir_num]

        if in_range(nr, nc):
            r, c = nr, nc
        else:
            dir_num = (dir_num + 2) % 4
    
    return r, c

ans_r, ans_c = func()
print(ans_r, ans_c)

File: 250808/작은 구슬의 이동/small-marble-movement.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [n, t] = input[0].split(" ").map(Number);
let [r, c, d] = input[1].split(" ");
[r, c] = [r, c].map(Number);

const grid = Array.from({ length: n }, () => Array(n).fill(0));

const dir = {
    'U': 0,
    'D': 3,
    'R': 1,
    'L': 2,
};

let dir_num = dir[d];

const [drs, dcs] = [[-1, 0, 0, 1], [0, 1, -1, 0]];

//
function oneMove(cr, cc) {
    const [nr, nc] = [cr + drs[dir_num], cc + dcs[dir_num]];

    if (!inRange(nr, nc)) {
        // 위치 변경
        dir_num = 3 - dir_num;
        return [cr, cc];
    }

    return [nr, nc];
}

function inRange(r, c) {
    return r >= 1 && r <= n && c >= 1 && c <= n;
}

function tMove(r, c) {
    let [currR, currC] = [r, c];

    for (let i = 0; i < t; i++) {
        [currR, currC] = oneMove(currR, currC);
    }

    return [currR, currC];
}

function printSpot(r, c) {
    const [resultR, resultC] = tMove(r, c);
    console.log(resultR, resultC);
}

printSpot(r, c);